### ISSUE TITLE
trilium: has two desktop files

### ISSUE THREAD

**Issue by mkg20001** [NixOS/nixpkgs#76567]
**Describe the bug**
The Trilium Desktop App has two desktop files

**To Reproduce**
1. Install trilium
2. Open the menu

**Expected behavior**
There should be one

**Screenshots**
![screenshot](https://user-images.githubusercontent.com/7735145/71497794-1c964800-285a-11ea-8f48-90d34ca9619c.png)

**Additional context**
/nix/store/y961ny44rfl33ismx3njads6y224lj7g-trilium-desktop-0.38.2
latest master

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.91, NixOS, 20.03pre-git (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(maciej): `"nixpkgs-20.03pre205792.fb1bc1b891f"`
 - channels(root): `"nixos-20.03pre206438.a06925d8c60"`
 - nixpkgs: `/home/maciej/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - trilium
# a list of nixos modules affected by the problem
module:
```

**Comment by worldofpeace** [comment]
Is there a desktop file at`~/ .local/share/applications`? The output of the derivation doesn't have multiple desktop files.

**Comment by mkg20001** [comment]
:sweat_smile: yeah. Removed, fixed

**Comment by worldofpeace** [comment]
@mkg20001 It's possible the program tries to install it based off some condition off FHS assumptions.

**Comment by mkg20001** [comment]
Has returned, FHS assumptions seem at play here

**Comment by matthewbauer** [comment]
It looks like an upstream bug, maybe not even specific to NixOS:

https://github.com/zadam/trilium/blob/62650a4545299f546f7b5f3bffb671103e2071b7/src/services/app_icon.js

**Comment by worldofpeace** [comment]
It seems they're trying to construct absolute paths from `ELECTRON_APP_ROOT_DIR` on a desktop file template and substitute those paths in https://github.com/zadam/trilium/blob/62650a4545299f546f7b5f3bffb671103e2071b7/src/services/app_icon.js#L51.
Weird, they do that and get around distributing a desktop file (and I think there's something to generate those in node). 

This does make it the intended function, and we just happen to ship with our own desktop file.
@matthewbauer It is a good report to send upstream.

**Comment by mkg20001** [comment]
Still not solved

**Comment by Myaats** [comment]
Solved by #204680
